### DISCUSSION TITLE
rke2 and kubelet ServiceMonitor

### DISCUSSION THREAD

**Discussion by uk1988** [rancher/rke2#5934] [Q&A]
As many, we are using Kube Prometheus Stack to deploy Prometheus and monitor our clusters.
When enabled Stack comes with build in ServiceMonitor for kubelet.

ServiceMonitor is looking for service over which it could connect to kubeleton nodes over port 10250. 
As I have seen there is no service in default installation of rke2 that gives access to kubelet.

Can we manually create a Service that would in k8s point to kubelet that is running on k8s nodes?

**Comment by brandond** [comment]
Sure? You can create whatever services you want. I'm not familiar with any Kubernetes distributions that create a Service for the kubelet checks though, usually the kubelet is checked by node monitors.
